ffi/src/scan.rs

```diff
@@ -499,19 +499,21 @@ struct ContextWrapper {
 #[no_mangle]
 pub unsafe extern "C" fn visit_scan_metadata(
     scan_metadata: Handle<SharedScanMetadata>,
+    engine: Handle<SharedExternEngine>,
     engine_context: NullableCvoid,
     callback: CScanCallback,
-) {
+) -> ExternResult<NullableCvoid> {
```

Comment:
I think `visit_scan_files` just returns its context. So I think we can either:
1. return that here (like you have as `NullableCvoid`) and add some docs and fix the `map` below OR
2. return `ExternResult<()>` instead and just use it to propagate the error, expecting engines to keep their context pointer around (like it would have been used before).

thoughts @nicklan? I tend to lean toward the second option for now just in the name of simplicity?